packages/gatsby-graphiql-explorer/src/app/snippets.js

```diff
@@ -0,0 +1,59 @@
+const getQuery = arg => {
+  const { operationDataList } = arg
+  const { query } = operationDataList[0]
+  return query
+}
+
+const pageQuery = {
+  name: `Page query`,
+  language: `JavaScript`,
+  codeMirrorMode: `jsx`,
+  options: [],
+  generate: arg => `import React from "react"
+import { graphql } from "gatsby"
+
+export default ({ data }) => <pre>{JSON.stringify(data, null, 4)}</pre>
+
+export const query = graphql\`
+${getQuery(arg)}
+\`
+`,
+}
+
+const staticHook = {
+  name: `StaticQuery hook`,
```

Comment:
Are the query names appropriate?

Reply:
I would use `useStaticQuery`, but I think having some note on what it is (and possibly link to documentation) would be great (I'm not sure if code-exporter has support to provide notes that are not part of code snippet?)

Reply:
Just checked, and it doesn't seem like we can provide notes with current version of `graphiql-code-exporter` so my thought is irrelevant. Let's keep your name - we can always change it later if we need to.